src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs

```diff
@@ -195,28 +209,28 @@ internal Type GetMonitorScaleMetricsTypeOrNull(IScaleMonitor monitor)
             return null;
         }
 
-        internal void LogStorageException(StorageException ex)
+        internal void LogStorageException(RequestFailedException ex)
         {
-            _logger.LogError(ex, $"An unhandled storage exception occurred when reading/writing scale metrics: {ex.ToString()}");
+            _logger.LogError(ex, $"An unhandled storage exception occurred when reading/writing scale metrics: {ex}");
```

Comment:
we shouldn't be including exceptions in the log message body. However, I am not sure if it is a breaking change to remove it.

If we do remove it, should format some properties from the `RequestFailedException` onto the message (using log templating, not string interpolation).

Reply:
An example of the exception message (when the failure comes from the service) would be this:

```
The table specified does not exist.
RequestId:0e0308a0-b002-0025-4595-e3385b000000
Time:2024-07-31T22:05:25.9397718Z
Status: 404 (Not Found)
ErrorCode: TableNotFound

Content:
{"odata.error":{"code":"TableNotFound","message":{"lang":"en-US","value":"The table specified does not exist.\nRequestId:0e0308a0-b002-0025-4595-e3385b000000\nTime:2024-07-31T22:05:25.9397718Z"}}}

Headers:
Transfer-Encoding: chunked
Server: Windows-Azure-Table/1.0 Microsoft-HTTPAPI/2.0
x-ms-request-id: 0e0308a0-b002-0025-4595-e3385b000000
x-ms-client-request-id: 7d830039-2628-46b8-91f5-1795b250ab34
x-ms-version: REDACTED
X-Content-Type-Options: REDACTED
Date: Wed, 31 Jul 2024 22:05:25 GMT
Content-Type: application/json;odata=minimalmetadata;streaming=true;charset=utf-8
```

Some of these values can be retrieved as `RequestFailedException` properties, such `Status `or `ErrorCode`. Others like RequestId or the specific message are part of the response `RequestFailedException.GetRawResponse()`: as properties 
like `RequestId`,  and others require to read the JSON content and parse it. `GetRawResponse` might return null when the failure is part of a batch.

More than happy to go with structured logging, my only concern is that we would need to be cautious to ensure that as part of exception processing we don't add too much overhead and also that we don't miss details when the reason is not a service issue itself: the inner could be `SocketException`, to name one example.